src/linechart.jsx

```diff
@@ -54,25 +60,122 @@ export default React.createClass({
         };
     },
 
-    renderLine() {
-        // Map series data to scaled points and filter to bounds of drawing area
-        const points = _.filter(
-            _.map(this.props.series.toJSON().points, d => {
-                return [this.props.timeScale(d[0]), this.props.yScale(d[1])];
-            }), p => this.inBounds(p));
+    renderLines() {
+        const pathLines = [];
+        let count = 1;
+        if (this.props.breakLine) {
+            // Remove nulls and NaNs from the line
+            let currentPoints = null;
+            _.each(this.props.series.toJSON().points, d => {
+                const value = d[1];
+                const badPoint = _.isNull(value) || _.isNaN(value) || !_.isFinite(value);
+                if (!badPoint) {
+                    if (!currentPoints) {
+                        currentPoints = [];
+                    }
+                    currentPoints.push([d[0], d[1]]);
+                } else {
+                    if (currentPoints) {
+                        const points = _.map(currentPoints,
+                            d => [this.props.timeScale(d[0]), this.props.yScale(d[1])]
+                        );
+                        if (points.length > 1) {
+                            pathLines.push(
+                                <path
```

Comment:
This `<path ...` construct occurs again below.  might be cleaner to encapsulate it in a function so it doesn't need to be updated in two places.